drivers/regulator/rpi-panel-attiny-regulator.c

```diff
@@ -37,11 +37,24 @@ static const struct regmap_config attiny_regmap_config = {
 static int attiny_lcd_power_enable(struct regulator_dev *rdev)
 {
 	unsigned int data;
+	int ret, i;
 
 	regmap_write(rdev->regmap, REG_POWERON, 1);
+	msleep(80);
+
 	/* Wait for nPWRDWN to go low to indicate poweron is done. */
-	regmap_read_poll_timeout(rdev->regmap, REG_PORTB, data,
-					data & BIT(0), 10, 1000000);
+	for (i = 0; i < 20; i++) {
```

Comment:
It's trying to work around dubious code in the Atmel - if it doesn't have time between commands then it can hard lock up whilst holding SDA low.

The REG_POWERON, 1 command does a load of processing from within the ISR, including delays, and bitbashing an I2C command to the TC358762 bridge chip. Hopefully the msleep covers that, but otherwise we poll a few times. It should never come anywhere near the 1s that it used to delay for.

As it happens I've never seen the PWRDNX line from the Toshiba go low, so a later patch removes this polling totally as part of switching away from REG_POWERON.

Reply:
So the increased delays are between attempts, not the total time for all attempts. That's fine.

Reply:
Syntax is ```regmap_read_poll_timeout(map, addr, val, cond, sleep_us, timeout_us)```, so it was polling every 10us up to a max of 1s. That's crazy fast, but also aborted should any of the transactions fail.

We're now sleeping for 80ms, then polling every 10ms for 20 attempts (ie 200ms max), so total time in the failure path actually comes down from 1s to 280ms (plus I2C transaction time).